New flow and UI for joining competition

Rcontre360 opened the pull request:
closes #967

Rcontre360 commented:
#967

ramirotw commented on apps/comps/components/user-agents/agent-card.tsx:
```diff
@@ -16,15 +16,18 @@ import { VerificationBadge } from "../verification-badge";
   ...props
 }) => {
   const router = useRouter();
+  console.log({ nameComp: !nameComponent || nameComponent === "link" });
```
delete this

ramirotw reviewed (changes requested):
1. values need to be wrapped in brackets [ ]
2. Prize Pool with the rewards is missing
3. title/description with body is not vertical aligned
4. custom Join button 
5. Registration deadline date format (lacks year)

<img width="737" height="654" alt="image" src="https://github.com/user-attachments/assets/5ada55f4-670c-4abb-858f-73173f6c2f9a" />

<img width="421" height="406" alt="image" src="https://github.com/user-attachments/assets/fd9c20d9-8f6c-4684-888d-f27d3bcf557c" />

ramirotw commented:
9. missing brackets

<img width="378" height="98" alt="image" src="https://github.com/user-attachments/assets/69875e84-8f3c-4c1c-a2ab-21f7b2a636ff" />

Rcontre360 commented:
> 1. values need to be wrapped in brackets [ ]
> 2. Prize Pool with the rewards is missing
> 3. title/description with body is not vertical aligned
> 4. custom Join button
> 5. Registration deadline date format (lacks year)
> 
> <img alt="image" width="737" height="654" src="https://private-user-images.githubusercontent.com/3328006/474057445-5ada55f4-670c-4abb-858f-73173f6c2f9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQzMTcwNTUsIm5iZiI6MTc1NDMxNjc1NSwicGF0aCI6Ii8zMzI4MDA2LzQ3NDA1NzQ0NS01YWRhNTVmNC02NzBjLTRhYmItODU4Zi03MzE3M2Y2YzJmOWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MDRUMTQxMjM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Y2FlMGNjMjQ5OGJkYTlmNmI4ZGM1ZjE2MjZkMWQ0YTA4MDA0MDJmYmJiZmNhMWEwZTBjYWU1NDRkNTcyNGU3ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.7Tg6LmO7ePy6ldNKPmw5WFFStOApytoNWZafqdLdEI4"> <img alt="image" width="421" height="406" src="https://private-user-images.githubusercontent.com/3328006/474058599-fd9c20d9-8f6c-4684-888d-f27d3bcf557c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQzMTcwNTUsIm5iZiI6MTc1NDMxNjc1NSwicGF0aCI6Ii8zMzI4MDA2LzQ3NDA1ODU5OS1mZDljMjBkOS04ZjZjLTQ2ODQtODg4ZC1mMjdkM2JjZjU1N2MucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MDRUMTQxMjM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWMwM2IwMjA4NzliMDI1NDQ1ZGJjMDViZWY0MDM0N2ZlNDNiNmMzNjE4ODgwZjg2OTU0MjFhZmFiMzlhNzI2OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.5bsxay9Et7pSHswlTjckwRDEA-_VGmDUf0S5cZbOyh0">

Currently im taking the reward from here @ramirotw 

<img width="697" height="242" alt="image" src="https://github.com/user-attachments/assets/5b1f86c7-d2b5-4693-86da-5ea51f260ba6" />

It doesnt appears because its undefined. Let me know where to take the reward value from

Rcontre360 commented:
> 9. missing brackets
> 
> <img alt="image" width="378" height="98" src="https://private-user-images.githubusercontent.com/3328006/474078767-69875e84-8f3c-4c1c-a2ab-21f7b2a636ff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQzMzQxNTYsIm5iZiI6MTc1NDMzMzg1NiwicGF0aCI6Ii8zMzI4MDA2LzQ3NDA3ODc2Ny02OTg3NWU4NC04ZjNjLTRjMWMtYTJhYi0yMWY3YjJhNjM2ZmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MDRUMTg1NzM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjE0Zjg4M2I4MDk1YzNkYWM0ZmFiNTJlMzc5MGQ3ZGU2ZWZhMTQ4NGI4MTk1NjQyZjk1MDJiN2VjMmYzYmUwYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.-awkO1nqhzTFPtTMU1aWjq8dc7Qd_Pkxkixo_MwkTvg">
> 10. we can filter out the agents that are already registered or disable them with some UI cue so the user knows which is registered and which is not. cc @dtbuchholz @jovirecall
> 
> <img alt="image" width="760" height="224" src="https://private-user-images.githubusercontent.com/3328006/474079541-6a39f48b-104f-4cc2-b1dd-be03165b44a2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQzMzQxNTYsIm5iZiI6MTc1NDMzMzg1NiwicGF0aCI6Ii8zMzI4MDA2LzQ3NDA3OTU0MS02YTM5ZjQ4Yi0xMDRmLTRjYzItYjFkZC1iZTAzMTY1YjQ0YTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgwNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MDRUMTg1NzM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTc4YzE1YjQ3NWE0NTJjNzUzNzBkMGZhOGQ3MzQ0M2I0NDMwMjYzZDhjOTg2MmQ5OGU4NmU1OWMyZjc3ZjZmNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cf8HiwV2n_0KoIfUg0tXQoCUE4WYYsoUGuiEKOq2z4k">

added a darker card, a tooltip and removed the pointer events when an agent is already selected

<img width="804" height="460" alt="image" src="https://github.com/user-attachments/assets/7cb04bd3-2185-473a-98c9-3230c3abbe9a" />

ramirotw commented:
@Rcontre360 now that https://github.com/recallnet/js-recall/pull/1007 is merged you can render the competition rewards using:
```jsx
<Rewards rewards={competition.rewards} compact />
```
<img width="854" height="835" alt="image" src="https://github.com/user-attachments/assets/3cea3a6d-1fe5-4662-9d57-2d163455d55d" />
